### PR TITLE
fix(preflights): template secrets for local preflights as well

### PR DESCRIPTION
**Pull request by ggrosseveurgg**
fix(preflights): template secrets for local preflights as well

### CODE REVIEW

**Review by ggguardian — CHANGES_REQUESTED** [review]

File: helm-preflights/preflights.sh [hunk 1]
```diff
@@ -172,8 +172,26 @@ then
...
+if [ -n "$PULL_SECRETS_OPTION" ] && [[ "$FORCE" == "yes" ]];
+then
+  echo -e "--- TEMPLATING PULL SECRETS"
+  helm template $NAMESPACE $VALUES_FILES $CHART_VERSION $PULL_SECRETS_OPTION $CHART > $script_dir/local_secrets.yaml 2>/dev/null
+  retcode_secret=$?
+  if [ $retcode_secret -ne 0 ];
+  then
+    echo_error "Unable to template pull secrets"
+    LOCAL_CHECKS_STATUS="error"
+    GLOBAL_RC=1
```
[the remark this answers is not in the document]
Comment (ggrosseveurgg) [reply helm-preflights/preflights.sh:184]: We would miss the fact that user played preflights in results/telemetry (we won't write anything to the gitguardian-preflights-results secret at the end). Not worth ?

[the remark this answers is not in the document]
Comment (ggguardian) [reply helm-preflights/preflights.sh:184]: Ok I see, but my feeling here is that it is not very easy to read/maintain, what about using the `trap` bash feature instead to run commands at the end of the script?

[the remark this answers is not in the document]
Comment (ggrosseveurgg) [reply helm-preflights/preflights.sh]: I rewrote part of the code to try to make it clearer

File: helm-preflights/preflights.sh [hunk 2]
```diff
@@ -142,7 +170,7 @@ while (("$#")); do
     if [ -n "$CHART" ];
     then
       usage
-      echo_error "You can have only one chart path, please check your command"
+      exit_error "You can have only one chart path, please check your command"
       exit 1
```
Comment (ggguardian) [inline helm-preflights/preflights.sh:174]: It can be removed now.

**Review by ggguardian — COMMENTED** [review]
LGTM 👍

**Pull request merged**